### ISSUE TITLE
Conditional content does not work with latex macros

### ISSUE THREAD

**Issue by lcnhb** [quarto-dev/quarto-cli#2580]
### Bug description

It seems like macros in hidden divs still get expanded by pandoc. This is problematic as I have defined some macros that only work with mathjax not latex (and vis versa) (for example the mathjax mathtip macro)

```quarto
::: {.content-visible unless-format="pdf"}

\newcommand{\test}{one}

:::

::: {.content-visible unless-format="html"}

\newcommand{\test}{two}

:::

$\test$
```

Quarto will complain that the second macro has already been defined irrespective of output. Ideally for macros one should be able to have a sort of preamble for html that gets parsed by pandoc (not just passed through like include-before-body does), thus one could specify which to include in yaml dependent on output wanted

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [ ] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
Yes, that's a fair bit annoying, unfortunately. The issue is that ultimately Pandoc is the other doing the interpretation of LaTeX commands for which you see the warning. You can test that in this way:

```
$ quarto render test-visible.qmd --to latex
```

The relevant result is:

```
\begin{document}
\maketitle
\ifdefined\Shaded\renewenvironment{Shaded}{\begin{tcolorbox}[interior hidden, sharp corners, boxrule=0pt, frame hidden, enhanced, borderline west={3pt}{0pt}{shadecolor}, breakable]}{\end{tcolorbox}}\fi

\newcommand{\test}{two}

\(one\)

\end{document}
```

Note how the latex source has the _expanded_ macro definition.

So you need to stop Pandoc from doing that. One (admittedly ugly) workaround is to wrap the relevant contents in raw blocks and leave the interpretation to happen at LaTeX compilation time (and HTML runtime). Do note the extra gymnastics needed:

````
---
title: test
keep-tex: true
---


::: {.content-visible unless-format="pdf"}

```{=html}
$$\newcommand{\test}{one}$$
```

:::

::: {.content-visible unless-format="html"}

```{=latex}
\newcommand{\test}{two}
```

:::

$\test$
````
